### PR TITLE
Fix various Ruff warnings in errors.py

### PR DESCRIPTION
**Pull request by JackKelly**
# Pull Request

## Description

* Tell `ruff` to also warn us about [`flake-8 bugbear` issues](https://pypi.org/project/flake8-bugbear/22.10.27/) (such as mutable default arguments in functions).
* Mutable arguments are a bit dangerous in Python :slightly_smiling_face:. Please see [this stack overflow discussion](https://stackoverflow.com/questions/26320899/why-is-the-empty-dictionary-a-dangerous-default-value-in-python) and [this web page](https://docs.quantifiedcode.com/python-anti-patterns/correctness/mutable_default_value_as_argument.html).
* Fix the mutable arguments in `errors.py`
* Fix missing description of `**kwargs` in some function docstrings.

Fixes #

## How Has This Been Tested?

- [x] Yes

## Checklist:

- [x] My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/.github/blob/main/coding_style.md)
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: ocf_ml_metrics/metrics/errors.py [hunk 1]
```diff
@@ -187,24 +197,24 @@ def compute_metrics(
     start_time: np.ndarray,
     filter_by_night: bool = False,
     tag: str = "",
-    thresholds: Optional[Union[list, float]] = -1,
```
Comment (JackKelly) [inline ocf_ml_metrics/metrics/errors.py:190]: `mypy` complained that `thresholds` must never be `None`. (And `Optional` is the same as `Union[..., None]`).

File: ruff.toml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 # Enable pycodestyle (`E`) and Pyflakes (`F`) codes by default.
-select = ["E", "F", "D", "I"]
+select = ["B", "E", "F", "D", "I"]
```
Comment (jacobbieker) [inline ruff.toml:2]: Should we make this the default for all repos? So update the template as well?

**Review by jacobbieker — APPROVED** [review]
LGTM!

File: ruff.toml (hunk 2 above)
Comment (jacobbieker) [reply ruff.toml:2]: Ah, just saw the other PR

**Pull request merged**